R/get_contact_matrix.R

```diff
@@ -0,0 +1,48 @@
+#' Create a contact matrix
+#'
+#' Function that loads and processes a contact matrix for use in the CoMo model
+#' ode solver. This function carries out a dimension change on the input
+#' contact matrix (from 16x16 to 21x21) to allow for the larger number of
+#' population categories provided by the population data
+#' (0-100+ years in 5 year intervals) compared to the contact data (0-80 in 5
+#' year intervals). This is done by repeating entries in the contact data
+#' corresponding to the oldest group (75-80 years) for the last population data
+#' age ranges (i.e. all 5 year categories for 80+ years).
+#'
+#' @param country String representing the country for which the contact matrix
+#' will be created. Naming conventions for countries must be in line with those
+#' used in input excel file.
+#' @param contact_type A named list of 16x16 matrices. Names are countries for
+#' which contact matrices are available. Each contact matrix represents the
+#' contact between different age groups in a given social setting e.g.
+#' contact_work gives contact matrix for the work place.
+#' @param n_5yr_agegroups_popstruc Number of 5 year age groups in the population
+#'  structure data. these ranges start with 0-4yrs and continue with 5 year gaps
+#'  up until the last category of 100+ years old. Default value set to 21.
+#'
+#' @return A square contact matrix for the specified country with dimension
+#' equal to \code{n_5yr_agegroups_popstruc}.
+#'
+get_contact_matrix <- function(country, contact_type
+                               , n_5yr_agegroups_popstruc = 21) {
+  #choose contact matrix for given country
+  c <- contact_type[[country]] %>% as.matrix()
+
+  #create a square matrix of 0s with dimension n_5yr_agegroups_popstruc
+  contact <- matrix(0, nrow = n_5yr_agegroups_popstruc,
```

Comment:
Can we name the rows and columns so that it's clear what element corresponds to what age combination? Think that might make it easier to understand further down where it's used in the code...

Reply:
Specifically, then you could do something like `contact_matrix["0-5", "75-80"]` to access a relevant age group.

Reply:
I've added names to rows and columns indicating the age ranges.

Reply:
Great, thanks.